tenseal/cpp/tensors/ckksvector.cpp

```diff
@@ -413,46 +494,60 @@ shared_ptr<CKKSVector> CKKSVector::enc_matmul_plain_inplace(
         this->add_inplace(tmp);
     }
 
-    this->_size = rows_nb;
```

Comment:
I don't think this method can work for multi-ciphertext vectors. You can see the rotation on line 492 which shouldn't work correctly in that case.